google/cloud/aiplatform/prediction/local_endpoint.py

```diff
@@ -128,13 +151,30 @@ def __init__(
             serving_container_environment_variables
         )
         self.serving_container_ports = serving_container_ports
-        self.credential_path = credential_path
-        self.host_port = host_port
-
         self.container_port = prediction_utils.get_prediction_aip_http_port(
             serving_container_ports
         )
 
+        self.credential_path = credential_path
+        self.host_port = host_port
+
+        self.gpu_count = gpu_count
+        self.gpu_device_ids = gpu_device_ids
+        self.gpu_capabilities = gpu_capabilities
+
+        if self.gpu_count and self.gpu_device_ids:
+            raise ValueError(
+                "At most one gpu_count or gpu_device_ids can be set but both are set."
+            )
+        if (self.gpu_count or self.gpu_device_ids) and self.gpu_capabilities is None:
```

Comment:
```suggestion
        if (self.gpu_count or self.gpu_device_ids) and not self.gpu_capabilities:
```

Do we want to check for empty list too? bool(list) returns False for both None and empty lists.

Reply:
It can be empty list.